### ISSUE TITLE
Android - map not always refreshing when moved

### ISSUE THREAD

**Issue by RyanTG** [bpoore/pbm-react#312]
This problem is in the current release as well as the current beta.

Notes:

Does not refresh when panning the map. But it DOES if your pan is a swipe. For example, does not refresh if you carefully move your thumb across the screen and then stop the pan by stopping your thumb. But it does refresh if you pan by flicking up, which causes the map to wander for a second before stopping.

Likewise with pinching to zoom in/out. A deliberate pinch results in no refresh. But a wild flick of a pinch will cause the map to move after you’ve lifted your fingers from the screen, resulting in a refresh.

Double tap to zoom results in refresh.

**Comment by RyanTG** [comment]
Things I found, which may or may not help:

https://github.com/react-native-maps/react-native-maps/issues/1540#issuecomment-412448644

https://github.com/react-native-maps/react-native-maps/issues/1557

**Comment by bpoore** [comment]
@RyanTG, I'll keep testing, but I believe [this](https://github.com/bpoore/pbm-react/tree/refactoring-map) is hopefully on the right track. I tried going down the path of removing the map coordinates from our global app state, but there were times when we did a search or a deep link where the map coordinates weren't aligned with the location data. Here, we are still keeping the coordinates in app-state, but refactored how we issue actions which set the coordinates. I can still occasionally make the map go nuts, but I feel like I have to try to get it to do so now and that it tends to happen when I'm fairly zoomed out?  Maybe I'm just being too hopeful that this is much better... but it feels like it is!

**Comment by RyanTG** [comment]
So far, yes this is looking swell!

I did get it to stutter here and there, but it stopped stuttering after a second. Like, it didn't get stuck in a stutter loop. Previously it would get stuck in an endless loop, so this is a major improvement. And I tried lots of different pan/zoom things (e.g. pan and then let go and then quickly pan again. things like that) and overall the experience was great. I'll keep playing with it, but initially impression is positive!! Nice work!
